flake8_trio.py

```diff
@@ -333,10 +353,30 @@ def has_exception(node: Optional[ast.expr]) -> str:
 
 
 class Visitor102(Flake8TrioVisitor):
+    class TrioScope:
```

Comment:
`get_matching_call` (previously `get_trio_scope`) no longer constructs a `TrioScope`, since only `Visitor102` actually cared about it. So moved it to a subclass

Reply:
IMO this isn't an affirmative reason *to* move it, so I'd keep the diff a bit smaller by leaving it where it was.  On the other hand, nor is there any real reason to move it *back*!

Reply:
haha. I liked subclassing it for people reading it later on not encountering it through a commit diff, fun tension